sentry_streams/sentry_streams/adapters/arroyo/reduce.py

```diff
@@ -66,6 +76,65 @@ def accumulator(self, result: Any, value: BaseValue[RoutedValue]) -> RoutedValue
         return routed
 
 
+class MetricsAccumulator(Accumulator[Message[Any], Any]):
+
+    def __init__(self, acc: Callable[[], Accumulator[Any, Any]], name: str) -> None:
+        self.acc = acc()
+        self.start_time: float | None = None
+        self.metrics = get_metrics()
+        self.tags = {"step": name}
+
+    def add(self, value: Message[Any]) -> Self:
+        if self.start_time is None:
+            self.start_time = time.time()
+        self.metrics.increment("streams.pipeline.input.messages", tags=self.tags)
+        self.metrics.increment(
+            "streams.pipeline.input.bytes", tags=self.tags, value=sys.getsizeof(value.payload)
```

Comment:
`getsizeof` feels a bit sketchy because the number doesn't mean anything for complex non-string types. i think it's probably better to check the type and get `len()`, and to not emit any metric if the payload isn't actually bytes